tools/ccm_extend.py

```diff
@@ -17,12 +18,28 @@
 headers = {'Authorization': 'Token {}'.format(ccm_token)}
 
 # Determine the cluster id of the cluster
-all_clusters = json.loads(requests.get(base_url+'/api/cluster/active/all', headers=headers).text)
-for cluster in all_clusters:
-    if cluster['name'] == cluster_name:
-        cluster_id = cluster['id']
+# It may take some time for CCM to have all of the clusters information. Poll
+# until it receives it. Try for 60 minutes, because we really don't want to waste a cluster we just created.
+give_up_time = time.time() + 60 * 60
+cluster_id = None
+while give_up_time > time.time():
+    print("Looking up cluster id.")
+    # Get all active clusters
+    all_clusters = json.loads(requests.get(base_url+'/api/cluster/active/all', headers=headers).text)
+    for cluster in all_clusters:
+        if cluster['name'] == cluster_name:
+            cluster_id = cluster['id']
+            print("Found cluster id!")
+            break
+
+    print("Cluster id not found. Sleeping for 60 seconds...")
```

Comment:
We do need a second break here tough, otherwise we're always going to wait 60 minutes. (the break above only breaks out of the `for` loop).